experimental/dds/tree2/src/feature-libraries/optional-field/optionalField.ts

```diff
@@ -316,132 +324,111 @@ export const optionalChangeRebaser: FieldChangeRebaser<OptionalChangeset> = {
 		revisionMetadata: RevisionMetadataSource,
 		existenceState?: NodeExistenceState,
 	): OptionalChangeset => {
-		const over = overTagged.change;
-
-		const perChildChanges = new ChildChangeMap<NodeChangeset>();
-		if (change.childChanges !== undefined) {
-			// TODO: (minor) early exits, better data structure choices, etc.
-			const overChildChanges = new ChildChangeMap<NodeChangeset>();
-			for (const [id, overChange] of over.childChanges ?? []) {
-				overChildChanges.set(id, overChange);
+		const withIntention = (id: RegisterId): RegisterId => {
+			if (id === "self") {
+				return id;
 			}
+			const intention = getIntention(id.revision ?? overTagged.revision, revisionMetadata);
+			return { revision: intention, localId: id.localId };
+		};
 
-			// If we're rebasing over a fieldChange, track ChangeAtomId_s for cases where a previously existing
-			// node was restored. In that case, when we construct our child id changes, they may apply to 'self' rather than
-			// the pre-existing childId.
-			let restoredRollbackChangeId: ChangeAtomId | undefined;
-			let restoredUndoChangeId: ChangeAtomId | undefined;
-			if (over.fieldChange !== undefined) {
-				const overIntention = getIntention(
-					over.fieldChange.revision ?? overTagged.revision,
-					revisionMetadata,
-				);
+		const { moves, childChanges, build } = change;
+		const { change: overChange } = overTagged;
+		const rebasedMoves: typeof moves = [];
+
+		const overDstToSrc = new ChildChangeMap<RegisterId>();
+		const overSrcToDst = new ChildChangeMap<RegisterId>();
+		for (const [src, dst] of overChange.moves) {
+			const srcTagged = withIntention(src);
+			const dstTagged = withIntention(dst);
+			overSrcToDst.set(srcTagged, dstTagged);
+			overDstToSrc.set(dstTagged, srcTagged);
+		}
 
-				if (over.fieldChange.newContent !== undefined) {
-					const overContent = over.fieldChange.newContent;
-					restoredRollbackChangeId = {
-						revision: overIntention,
-						localId: over.fieldChange.id,
-					};
-					if ("revert" in overContent) {
-						restoredUndoChangeId = overContent.revert;
-					}
-				}
+		const renamedDsts = new ChildChangeMap<RegisterId>();
+		for (const [_, dst] of moves) {
+			renamedDsts.set(dst, dst);
+		}
+		for (const [src, dst] of overSrcToDst.entries()) {
+			if (!renamedDsts.has(src)) {
+				renamedDsts.set(src, dst);
 			}
+		}
```

Comment:
I must say I'm having trouble understanding what `renamedDsts` represents. Which pair of contexts is this spanning?
It's going away fairly soon though, so I'm okay with it.

Reply:
Added this explanation and moved this to be more colocated with where it's used:

```
		// Maps the content occupying a given register id in the output context of `change` to the register id
		// that content occupies in the output context of the rebased change.
		// This is necessary since child changes to nodes are keyed on the register they occupy in the output
		// context of the changeset, and `change` might make child changes to content which it doesn't move, but
		// `over` does.
```